components/Main.js

```diff
@@ -0,0 +1,123 @@
+import React, { useEffect, useState } from "react";
+import Constants from "expo-constants";
+import * as Location from "expo-location";
+import styled from "styled-components/native";
+import Loading from "./Loading";
+import { Platform } from "react-native";
+
+const Main = () => {
+  const [heading, setHeading] = useState(null);
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  // The user has to give permission to the app to use the location when the app is opened.
+  // If permission is not given the user gets a error message and otherwhise the Location data is fetched
+  // Also set at setTimeout of 2 seconds on the actions which triggers a longer loading screen.
+  useEffect(async () => {
+    let { status } = await Location.requestForegroundPermissionsAsync();
+    setTimeout(() => {
+      if (status !== `granted`) {
+        setErrorMessage({
+          errorMessage: `Permission to access location was denied!`,
+        });
+      } else {
+        Location.watchHeadingAsync((data) => {
+          setHeading(Math.round(data.trueHeading));
+        });
+      }
+    }, 2000);
+  }, []);
+
+  // Sets which compass text will be shown depending on the heading from the phone.
+  const getCompassText = (heading) => {
+    if ((heading >= 337 && heading <= 360) || (heading >= 0 && heading < 22)) {
+      return "North";
+    } else if (heading >= 22 && heading < 67) {
+      return "Northeast";
+    } else if (heading >= 67 && heading < 112) {
+      return "East";
+    } else if (heading >= 112 && heading < 157) {
+      return "Southeast";
+    } else if (heading >= 157 && heading < 202) {
+      return "South";
+    } else if (heading >= 202 && heading < 247) {
+      return "Southwest";
+    } else if (heading >= 247 && heading < 292) {
+      return "West";
+    } else if (heading >= 292 && heading < 337) {
+      return "Northwest";
```

Comment:
Interesting to see how you implemented the cardinal points in a function! I was wondering if this could be expressed as a switch statement?

Reply:
I thought about doing it with a switch but decided against it since i am doing the statement inbetween different values. I was unsure if switch could be used for this purpose in a easy way.

Reply:
I see! Yeah, I haven't tried switch for that purpose either.